tests/clickhouse-test

```diff
@@ -1248,39 +1325,35 @@ class TestCase:
 
         return None
 
-    def process_result_impl(
-        self, proc, stdout: str, stderr: str, debug_log: str, total_time: float
-    ):
+    def process_result_impl(self, proc, total_time: float):
+        if proc:
+            if proc.returncode is None:
+                kill_process_group(os.getpgid(proc.pid))
+
         description = ""
 
+        debug_log = ""
+        if os.path.exists(self.testcase_args.debug_log_file):
+            with open(self.testcase_args.debug_log_file, "rb") as stream:
+                debug_log += self.testcase_args.debug_log_file + ":\n"
+                debug_log += str(stream.read(), errors="replace", encoding="utf-8")
+                debug_log += "\n"
+
+        stdout = ""
+        if os.path.exists(self.stdout_file):
+            with open(self.stdout_file, "rb") as stdfd:
+                stdout = str(stdfd.read(), errors="replace", encoding="utf-8")
+
+        stderr = ""
+        if os.path.exists(self.stderr_file):
+            with open(self.stderr_file, "rb") as stdfd:
+                stderr += str(stdfd.read(), errors="replace", encoding="utf-8")
+
         if debug_log:
             debug_log = "\n".join(debug_log.splitlines()[:100])
 
         if proc:
             if proc.returncode is None:
-                try:
-                    pgid = os.getpgid(proc.pid)
-                    # NOTE: this still may leave some processes, that had been
-                    # created by timeout(1), since it also creates new process
-                    # group. But this should not be a problem with default
-                    # options, since the default time for each test is 10min,
-                    # and this is way more bigger then the timeout for each
-                    # timeout(1) invocation.
-                    #
-                    # But as a workaround we are sending SIGTERM first, and
-                    # only after SIGKILL, that way timeout(1) will have an
-                    # ability to terminate childrens (though not always since
-                    # signals are asynchronous).
-                    os.killpg(pgid, signal.SIGTERM)
-                    # This may not be enough, but this is at least something
-                    # (and anyway it is OK to spend 0.1 second more in case of
-                    # test timeout).
-                    sleep(0.1)
-                    os.killpg(pgid, signal.SIGKILL)
-                except OSError as e:
-                    if e.errno != ESRCH:
-                        raise
-
                 if stderr:
```

Comment:
Throwing a question to the air:

Are we not including in this branch `stdout` to `description` on purpose?
The inclusion of `stdout` seems missing here, while there are some redundant collections in the next branch (line 1395) and even some duplicated addition to `description` since it's in line 1413 but then also in 1441. Maybe this is the intending behavior, but I find it confusing

Reply:
This code requires some cleanup.

> Are we not including in this branch stdout to description on purpose?

I don't think that `clickhouse-test` wrapper has some code that is "on purpose", but does not fills logical, and don't have comments for this "smart logic".

That said that I would say that 99% it is a bug in the code.

But let's address it separately, since the purpose of this PR is different.

>while there are some redundant collections in the next branch (line 1395) and even some duplicated addition to description since it's in line 1413 but then also in 1441. Maybe this is the intending behavior, but I find it confusing

About the case when stdout included twice - once if there is stderr, another if there is `@@SKIP@@` in stdout - I think it has been written with the assumption that SKIP cannot fail, and I guess it is "almost always" true